web/src/user_group_pill.ts

```diff
@@ -26,18 +37,18 @@ export function create_item_from_group_name(group_name, current_items) {
     return item;
 }
 
-export function get_group_name_from_item(item) {
+export function get_group_name_from_item(item: PillItem): string {
     return item.group_name;
 }
 
-function get_user_ids_from_user_groups(items) {
+function get_user_ids_from_user_groups(items: PillItem[]): number[] {
     const group_ids = items.map((item) => item.id).filter(Boolean);
     return group_ids.flatMap((group_id) => [
         ...user_groups.get_user_group_from_id(group_id).members,
     ]);
 }
 
-export function get_user_ids(pill_widget) {
+export function get_user_ids(pill_widget: {items: () => PillItem[]}): number[] {
```

Comment:
`pill_widget: PillWidget`?

Reply:
got a bit confused, working on it though